### PR TITLE
test: Cover error handling cases in API tests (1/2)

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Tidies up a few uncovered branches with test cases following refactors for documentation and stronger enforcing of types
 - Most test cases just cover exception handling, but there's a few others thrown in
 - I see it as a helpful step before trying to take on https://trello.com/c/TfmJiwID/2686-fix-airbrake-error-logging-on-api
 - PR was growing in size so I'll pick up the rest next week - happy for this to be merged upon review

### API test coverage change in https://github.com/theopensystemslab/planx-new/pull/2466 & https://github.com/theopensystemslab/planx-new/pull/2482
| - | Before | After |
|--------|--------|--------|
| Statements | 79.32% | 80.41% | 
| Branches | 50.69% | 55.68% | 
| Functions | 67.88% | 67.88% | 
| Lines | 78.79% | 79.66% |

### CODE REVIEW

File: api.planx.uk/modules/flows/copyFlowAsPortal/service.ts [hunk 1]
```diff
@@ -7,7 +7,6 @@ import { Flow } from "../../../types";
 const copyPortalAsFlow = async (flowId: string, portalNodeId: string) => {
   // fetch the parent flow data
   const flow = await getFlowData(flowId);
-  if (!flow) throw Error("Unknown flowId");
```
Comment (DafyddLlyr) [inline api.planx.uk/modules/flows/copyFlowAsPortal/service.ts:10]: This was redundant and unreachable, this exception is already thrown inside `getFlowData()`

File: api.planx.uk/modules/send/email/index.ts [hunk 2]
```diff
@@ -44,7 +45,7 @@ export async function sendToEmail(
     // Prepare email template
     const config: EmailSubmissionNotifyConfig = {
       personalisation: {
-        serviceName: flowName || "PlanX",
```
Comment (DafyddLlyr) [inline api.planx.uk/modules/send/email/index.ts:47]: This was also unreachable - all flows must have names.

File: api.planx.uk/modules/send/email/index.ts [hunk 3]
```diff
@@ -54,12 +55,7 @@ export async function sendToEmail(
 
     // Send the email
     const response = await sendEmail("submit", sendToEmail, config);
-    if (response?.message !== "Success") {
```
Comment (DafyddLlyr) [inline api.planx.uk/modules/send/email/index.ts:57]: Again, this code was unreachable - `message` is only set to `"Success"` and nothing else.

File: api.planx.uk/lib/notify/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  });
+
+  it("throw an error if the NotifyClient errors", async () => {
+    const mockNotifyClient = NotifyClient.mock.instances[0];
+    mockNotifyClient.sendEmail.mockRejectedValue({
+      response: { data: { errors: ["Invalid email"] } },
+    });
+    await expect(sendEmail("save", TEST_EMAIL, mockConfig)).rejects.toThrow();
+  });
+});
```
Comment (jessicamcinchak) [inline api.planx.uk/lib/notify/index.test.ts:39]: These are super nice & clear :raised_hands:

**Review by jessicamcinchak — APPROVED** [review]

**Pull request merged**